dali/operators/reader/loader/coco_loader.cc

```diff
@@ -483,6 +486,26 @@ void CocoLoader::ParseJsonAnnotations() {
   detail::ParseJsonFile(spec_, image_infos, annotations, category_ids,
                         parse_segmentation, output_pixelwise_masks_);
 
+  if (!images_.empty()) {  // Filter for image files was provided.
+    std::unordered_set<int> ids_to_keep;
+    image_infos.erase(
+      std::remove_if(image_infos.begin(), image_infos.end(),
+        [&](const detail::ImageInfo &info) {
+          if (images_.find(info.filename_) == images_.end()) {
```

Comment:
It could be very inefficient. Call find for every sample.
At least you can sort it and bisect.

Reply:
Nvm